### PR TITLE
TrollApps 1.3

### PR DESCRIPTION
**Pull request by Bonnie39**
I made a few modifications to the Sources and Featured view to mitigate crashes that would occur in specific scenarios in which the user did not have a working internet connection. I also updated the featured view to use a .json file hosted in my fork repository rather than from haxi0's TrollApps Static API repo so that I could fix a few missing icons on the Featured page (image links were broken since they were originally hosted on Discord).

To summarize this update:
*Fixes a crash that occurs when attempting to view the Sources tab without an internet connection
*Fixes a crash that occurs when attempting to download an app from the Featured tab without an internet connection
*Now displays an error message on the Featured tab if a download fails

### CODE REVIEW

**Comment by Bonnie39** [comment]
wen merge?
